### DISCUSSION TITLE
Subquery aggregation error

### DISCUSSION THREAD

**Discussion by EvgSkv** [prestodb/presto#16010] [Q&A] [answered]
Hello,
I've encountered an error for a simple query that I needed. Here is a simplified example that still produces error:

```
presto> select x, (select array_agg(x + y) from unnest(sequence(0, 10)) as t(y)) from unnest(sequence(0, 10)) as z(x);
Query 20210427_055016_00015_swnyx failed: line 1:12: Given correlated subquery is not supported
```

Is there some configuration that I could enable to make this query work?

Thank you!

**Comment by bitsondatadev** [comment, marked as answer]
Unfortunately, that feature is not supported in prestodb. There’s no configuration option, as it reflects a limitation in the query planner and optimizer.

**Comment by oliverjacob73** [comment]
Looks like this works in [Trino](https://trino.io/) though. Trino is another query engine that is practically the same as Presto but [split off a couple years back](https://trino.io/blog/2020/12/27/announcing-trino.html). 

```
trino:tiny> select x, (select array_agg(x + y) from unnest(sequence(0, 10)) as t(y)) from unnest(sequence(0, 10)) as z(x);
 x  |                    _col1
----+----------------------------------------------
  0 | [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
  1 | [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
  2 | [2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
  3 | [3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13]
  4 | [4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14]
  5 | [5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
  6 | [6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]
  7 | [7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17]
  8 | [8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18]
  9 | [9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]
 10 | [10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
(11 rows)
```

**Comment by rohanpednekar** [comment]
Hey @EvgSkv, We are actively working on https://github.com/prestodb/presto/pull/15575 which may fix this issue.  Please stay tuned! ✌️

**Comment by bitsondatadev** [comment]
![Screen Shot 2021-06-09 at 11 53 47 AM](https://user-images.githubusercontent.com/8547669/121388284-5b6a7180-c919-11eb-8041-635e8956d064.png)
Typical bullying of the Presto community. Why the thumbs down @rohanpednekar? Also #15575 doesn't solve the issue @EvgSkv is referencing. 

@oliverjacob73 one thing you should be aware of is that [there is a policy](https://github.com/prestodb/foundation/blob/main/COMMUNICATION-POLICY.md) that is set so that you're not allowed to talk about other open source projects like Trino. I think that's why @rohanpednekar downvoted your answer. Just FYI, don't post about it here or you can get kicked off.

**Reply by oliverjacob73** [reply]
Thanks @bitsondatadev, wasn't aware. I am curious why Presto people don't want us to talk about another project (not sure if I should say the name?) if it solves the issue. I was just trying to help really.

**Comment by EvgSkv** [comment]
@bitsondatadev @rohanpednekar @oliverjacob73 
Thank you for elaboration! I'll stay tuned.
